front/components/items/AppointmentApproval/index.tsx

```diff
@@ -0,0 +1,146 @@
+import Headline from '../Headline';
+import Paragraph from '../Paragraph';
+import { AppointmentDataContext } from 'components/wrappers/AppointmentDataContext';
+import Button from '../Button';
+import { useRouter } from 'next/router';
+import { strapiAppointmentPut } from 'lib/strapi/appointments/put';
+import { findProperAppointmentCode, giveAppointmentCode } from './helper';
+import { strapiAppointmentGet } from 'lib/strapi/appointments/get';
+import { strapiAppointmentQuery } from 'lib/strapi/appointments/queryType';
+
+const AppointmentApproval = () => {
+    // TODO: fix overflow after leaving page
+    const router = useRouter();
+    const [isDeclined, setIsDeclined] = useState(false);
+    const [isApproved, setIsApproved] = useState(false);
+    const [appointments, getAppointments] = useState<strapiAppointmentQuery[]>();
+    useEffect(() => {
+        strapiAppointmentGet().then((res) => {
+            getAppointments(res.data.data);
+        });
+    }, [isApproved]);
+    const { appointmentID } = useContext(AppointmentDataContext);
+    const bookingAppointment = {
+        data: {
+            is_booked: true,
+            appointment_code: giveAppointmentCode(appointments),
+        },
+    };
+    return (
+        <div className="appointmentapproval">
+            <div className="appointmentapproval-alert">
+                {appointmentID === undefined ? (
+                    <React.Fragment>
+                        <Headline
+                            variant="h1"
+                            text="Nie powinno Cię tu być, ale..."
+                            colorClass="greendark"
+                            placeClass="center"
+                        />
+                        <Paragraph
+                            size="medium"
+                            text="Nie martw się! Będziesz mógł tu wrócić, gdy zarezerwujesz wizytę u swojego terapuety."
+                            colorClass="greendark"
+                        />
+                        <Button
+                            variant="h3"
+                            colorClass="greendark"
+                            text="Umów wizytę"
+                            isLink
+                            link="/contact"
+                        />
+                    </React.Fragment>
+                ) : (
+                    <React.Fragment>
+                        <Headline
+                            variant="h1"
+                            text={`${
+                                (isApproved && 'Potwierdzono') ||
+                                (isDeclined && 'Anulowano') ||
+                                'Potwierdzenie wizyty'
+                            }`}
+                            colorClass="greendark"
+                            placeClass="center"
+                        />
+                        {/* TODO print appointment_code */}
+                        {[appointmentID]?.map((item, index) => {
```

Comment:
is it array with mnore than one value? I'd rename it to appointmentIDs
or maybe to appointmentsIDs
I don't know how to do it in correct english xD